amazon/ion/core.py

```diff
@@ -446,53 +461,46 @@ def __new__(cls, *args, **kwargs):
             del kwargs[TIMESTAMP_PRECISION_FIELD]
         if TIMESTAMP_FRACTION_PRECISION_FIELD in kwargs:
             fractional_precision = kwargs.get(TIMESTAMP_FRACTION_PRECISION_FIELD)
-            if fractional_precision is not None and fractional_precision < 1:
+            if fractional_precision is not None and 1 > fractional_precision > MICROSECOND_PRECISION:
```

Comment:
When `microseconds == 0`, why wouldn't `fractional_precision == 1` be allowed?  That is to say, When `fractional_precision == None`, then it defaults to effectively `fractional_precision == 6` right?

In other words, what is the precision of `microseconds` when this is not specified (as is the case with a normal `datetime`)?

Reply:
The logic exists where when  `fractional_precision == None` then `fractional_precision` defaults to 6, but this exists in the `timestamp` method on line [548](https://github.com/amzn/ion-python/pull/100/files#diff-6328c9946663c2b27d43e85f0fa42c3fR548).
Unfortunately, with the way unit tests are set up, for some tests if data just goes through the `Timestamp` constructor with `fractional_precision` being originally `None`, then the test expects the newly constructed `Timestamp` object to have `fractional_precision` of `None`. But if the data go through the `timestamp` function with `fractional_precision` being originally `None`, then the test expects to see `fractional_precision == 6`.
If I try moving the logic to make `fractional_precision == 6` from `timestamp` to `Timestamp`, unit tests start to fail due to the aforementioned reason.